### PR TITLE
impl: bootstrap Timestamp type

### PR DESCRIPTION
**Pull request by coryan**

### CODE REVIEW

File: types/src/timestamp.rs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        let odt = time::OffsetDateTime::parse(&value, &Rfc3339).map_err(E::custom)?;
+        let nanos_since_epoch = odt.unix_timestamp_nanos();
+        let seconds = (nanos_since_epoch / NS) as i64;
+        let nanos = (nanos_since_epoch % NS) as i32;
+        Ok(Self::Value {
+            seconds: seconds,
```
Comment (codyoss) [inline types/src/timestamp.rs:99]: nit: clippy warning


```suggestion
            seconds,
```

File: types/src/timestamp.rs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            nanos: nanos,
```
Comment (codyoss) [inline types/src/timestamp.rs:100]: nit: clippy warning


```suggestion
            nanos,
```

File: types/src/timestamp.rs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("a string with a timestamp in RFC 3339 format")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        let odt = time::OffsetDateTime::parse(&value, &Rfc3339).map_err(E::custom)?;
```
Comment (codyoss) [inline types/src/timestamp.rs:94]: nit: clippy warning


```suggestion
        let odt = time::OffsetDateTime::parse(value, &Rfc3339).map_err(E::custom)?;
```

**Review by codyoss — APPROVED** [review]
LGTM, maybe we should run clippy in CI? Not sure other code is compliant either though.

**Review by coryan — COMMENTED** [review]
Enabling clippy sounds like a good idea.  We can first scope it to just `types` and add `google-cloud-auth` later.

**Pull request merged**